### DISCUSSION TITLE
Google Sheets doc stored on Google Drive used as master data base?

### DISCUSSION THREAD

**Discussion by fritzisa** [code4recovery/12-step-meeting-list#681] [Q&A] [answered]
**Dear friends,**
We recently launched our new site for [Area 59](https://area59aa.org) (Eastern PA).
Almost all of our Districts and Intergroups have provided me with a link to their data.

I circulated the attached PDF back when I was asking for help from our contributing websites.
Recently, one of our Districts told me their Google Sheets doc (see attached .xlsx file) serves as their database, provided it's stored on Google Drive. I understand how to use their spreadsheet as a CSV file and import it into our meetings list.
But from my understanding, if they make changes to their file they would need to send me an updated file for re-import?
They told me their file is connected to GSO and any changes they make are reflected in GSO's meeting list.
I can't find any documentation to confirm or deny this process.

Can you please provide some clarification on what can and can't be done with a spreadsheet similar to theirs?

[Meeting List Template Williamsport PA - Meetings.xlsx](https://github.com/code4recovery/12-step-meeting-list/files/8051929/Meeting.List.Template.Williamsport.PA.-.Meetings.xlsx)

[Area-59-Meetings-HELP-Request-NOV-11-2021.pdf](https://github.com/code4recovery/12-step-meeting-list/files/8051975/Area-59-Meetings-HELP-Request-NOV-11-2021.pdf)

Thanks in advance for your assistance!

In fellowship,
**Scott F.**  |  Area 59 Web Servant

**Comment by kiyote33** [comment]
Hi @fritzisa, If you are asking about how the 12 Step Meeting List plugin uses data from a Google Sheets doc, I can advise that it is normally imported through the data source feature on the Import & Settings page via a link, much like a JSON data import. All the imported data source  records are stored locally in the MySql database, but we suggest that data source records should not be updated directly. To update these local records, you should receive updates from the data source by refreshing the Data Source link.  

You would probably be best served to ask the Williamsport website to share their website Meeting List data with Area 59 like everyone else (per the instructions you provided in the attached PDF).  Then when you refresh the associated data source link you will be in sync with the meeting information, much the same as GSO's Meeting Guide app.  There's no need to send you a copy of their Google Sheets doc every time an update is made.  

Hope this helps. Please advise if we can be a further assistance.

**Comment by fritzisa** [comment]
Thank you Wayne,

I neglected to mention that this District’s site isn’t built in Wordpress.

My PDF instructions were for those Districts and Intergroups who are already using TSML.

We still have (3) Districts that aren’t using Wordpress and TSML, and this is one of them.

They told me they provided GSO with this Google Sheets doc. and they make edits to their file and somehow that is automatically updated on GSO’s listing.

I don’t understand how that would work.

I’m trying to confirm if using occasionally updated CSV files, is the best alternative for these (3) remaining Districts.

Please let me know if I’m missing something here.

I’ve looked unsuccessfully for documentation addressing this topic.
 
Thank you for your service!

In fellowship,

Scott Fritzinger  |  Area 59 Web Servant

**Reply by kiyote33** [reply]
Yes, using updated Google Sheets linked as a Data Source would be the simplest way to occasionally update those districts. A more manual step would be too convert the Google Sheets spreadsheet to a CSV file and manually import it, but I would suggest you would be better off just linking the updated Google Sheets doc and let TSML handle the reformatting.

**Comment by fritzisa** [comment]
Hi Wayne,

Sorry for my ignorance.

The Google Sheet I was provided has this link for sharing:

https://docs.google.com/spreadsheets/d/1jhAO6X9egDa18Pp04n3NcLp9_CM9gcEu0IVUSCZG0OQ/edit?usp=sharing <https://docs.google.com/spreadsheets/d/1jhAO6X9egDa18Pp04n3NcLp9_CM9gcEu0IVUSCZG0OQ/edit?usp=sharing>

When I paste in that link as the Data Source I get the error message:  JSON: Syntax error, malformed JSON.
Can you tell what am I doing wrong?
Thank you,



Scott Fritzinger  |  Area 59 Web Servant

**Comment by kiyote33** [comment, marked as answer]
We have a new utility for importing Google Sheets at **https://sheets.code4recovery.org**.  I ran your link through it and it generated the following link: https://sheets.code4recovery.org/storage/1jhAO6X9egDa18Pp04n3NcLp9_CM9gcEu0IVUSCZG0OQ.json which I then tested and it loaded successfully.  Our Google Sheets project at sheets.code4recovery.org fetches meeting data from Google Sheets using the v4 API and transforms it into a CORS-enabled minified Meeting Guide-spec JSON file.  So you can use it to do a data source import. 

Hope this works for you. Let me know if there is anything else I can do to help.

**Reply by joshreisner** [reply]
just know it needs to be refreshed every time theres a change

**Reply by kiyote33** [reply]
Alternately - if Williamsport provides you with the same exact importer link that they've provided to GSO for a data source, then you wouldn't need to be wary of needing to do a two-step refresh process for Williamsport: (1. Refresh the **Importer Link at https://sheets.code4recovery.org**; 2. Refresh the local **Import & Settings** page "Williamsport" data source). It remains the responsibility of Williamsport to refresh the Importer link at https://sheets.code4recovery.org which they have to do anyway so GSO gets their updates.

**Comment by fritzisa** [comment]
Thanks so much Wayne, that did the trick.

I’ll save the resources you’ve provided.

Have great weekend!

In fellowship,

Scott Fritzinger  |  Area 59 – Web Servant

**Comment by fritzisa** [comment]
Thanks Josh,

We’ve got about a dozen feeds that we refresh ~ every other day.

In fellowship,

Scott Fritzinger  |  Area 59 Web Servant  |  (610) 762-7792

**Comment by joshreisner** [comment]
right and now there's another step. to get the latest data from the google sheet, visit this address so the feed is refreshed https://sheets.code4recovery.org/1jhAO6X9egDa18Pp04n3NcLp9_CM9gcEu0IVUSCZG0OQ

then refresh it in wordpress.

(this workflow wasn't designed for this use-case, which is why it's a little clunky)

**Comment by fritzisa** [comment]
Got it, thank you for the heads up!
